src/ja/worker/config.py

```diff
@@ -10,27 +9,73 @@ class WorkerConfig(Config):
     Config for the worker client.
     """
 
+    def __init__(self, uid: str, ssh_config: SSHConfig, resource_allocation: ResourceAllocation):
+        """!
+        creates resource allocation instance
+        """
+        if uid is None:
+            raise ValueError("missing uid")
+        if ssh_config is None:
+            raise ValueError("missing ssh config")
+        if resource_allocation is None:
+            raise ValueError("missing resources")
+
+        self._ssh_config = ssh_config
+        self._resource_allocation = resource_allocation
+        self._uid = uid
+
+    def __eq__(self, o: object) -> bool:
+        if isinstance(o, WorkerConfig):
+            return self._ssh_config == o._ssh_config \
+                and self._uid == o._uid \
+                and self._resource_allocation == o._resource_allocation
+        else:
+            return False
+
     @property
     def uid(self) -> str:
         """!
         @return: The desired UID to register with on the server.
         """
+        return str(self._uid)
 
     @property
     def ssh_config(self) -> SSHConfig:
         """!
         @return: Parameters for establishing an ssh connection to the server.
         """
+        return self._ssh_config
 
     @property
     def resources(self) -> ResourceAllocation:
         """!
         @return: The resources available on this work machine.
         """
+        return self._resource_allocation
 
     def to_dict(self) -> Dict[str, object]:
-        pass
+        d: Dict[str, object] = dict()
+        d["ssh_config"] = self._ssh_config.to_dict()
+        d["resource_allocation"] = self._resource_allocation.to_dict()
+        d["uid"] = self._uid
+        return d
 
     @classmethod
     def from_dict(cls, property_dict: Dict[str, object]) -> "WorkerConfig":
-        pass
+        if "resource_allocation" in property_dict:
```

Comment:
`cls._get_dict_from_dict` automatically raises a ValueError if the dictionary does not contain the requested key.
There is no need for you to manually check if property_dict contains a key beforehand.